kernel/kernel-api/src/main/java/io/delta/kernel/internal/CommitRangeImpl.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright (2025) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.delta.kernel.internal;
+
+import static io.delta.kernel.internal.util.Preconditions.checkArgument;
+import static java.util.Objects.requireNonNull;
+
+import io.delta.kernel.CommitRange;
+import io.delta.kernel.CommitRangeBuilder;
+import io.delta.kernel.Snapshot;
+import io.delta.kernel.data.ColumnarBatch;
+import io.delta.kernel.engine.Engine;
+import io.delta.kernel.internal.annotation.VisibleForTesting;
+import io.delta.kernel.internal.fs.Path;
+import io.delta.kernel.utils.CloseableIterator;
+import io.delta.kernel.utils.FileStatus;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+/** Implementation of {@link CommitRange}. */
+public class CommitRangeImpl implements CommitRange {
+
+  private final Path dataPath;
+  private final Optional<CommitRangeBuilder.CommitBoundary> startBoundaryOpt;
+  private final Optional<CommitRangeBuilder.CommitBoundary> endBoundaryOpt;
+
+  private final long startVersion;
+  private final long endVersion;
+  // TODO: update contents of this class for CCV2 tables
+  //  (include ratified commits? store a LogSegment?)
+  private final List<FileStatus> deltaFiles;
+
+  public CommitRangeImpl(
+      Path dataPath,
+      Optional<CommitRangeBuilder.CommitBoundary> startBoundaryOpt,
+      Optional<CommitRangeBuilder.CommitBoundary> endBoundaryOpt,
+      long startVersion,
+      long endVersion,
+      List<FileStatus> deltaFiles) {
```

Comment:
should we validate that the first delta file = start? and last delta file = end?

Reply:
Personally I'm not really sure how much validation we should do here, I'm open to opinions. All input here should in theory already be validated by the functions that generated it.

If we validate that, I'd also wonder if we should check for contiguity as well